openverse_catalog/dags/providers/provider_api_scripts/provider_data_ingester.py

```diff
@@ -101,23 +144,89 @@ def ingest_records(self, **kwargs) -> None:
         logger.info(f"Begin ingestion for {self.__class__.__name__}")
 
         while should_continue:
-            query_params = self.get_next_query_params(query_params, **kwargs)
-
-            batch, should_continue = self.get_batch(query_params)
-
-            if batch and len(batch) > 0:
-                record_count += self.process_batch(batch)
-                logger.info(f"{record_count} records ingested so far.")
-            else:
-                logger.info("Batch complete.")
-                should_continue = False
+            query_params = self.get_query_params(query_params, **kwargs)
+            if query_params is None:
+                # Break out of ingestion if no query_params are supplied. This can
+                # happen when the final `manual_query_params` is processed.
+                break
+
+            try:
+                batch, should_continue = self.get_batch(query_params)
+
+                if batch and len(batch) > 0:
+                    record_count += self.process_batch(batch)
+                    logger.info(f"{record_count} records ingested so far.")
+                else:
+                    logger.info("Batch complete.")
+                    should_continue = False
+
+            except Exception as error:
+                ingestion_error = IngestionError(
+                    error, traceback.format_exc(), query_params
+                )
+
+                if self.skip_ingestion_errors:
+                    # Add this to the errors list but continue processing
+                    self.ingestion_errors.append(ingestion_error)
+                    logger.error(f"Skipping batch due to ingestion error: {error}")
+                    continue
+
+                # Rethrow the exception so the taskrun fails
+                raise error from ingestion_error
 
             if self.limit and record_count >= self.limit:
                 logger.info(f"Ingestion limit of {self.limit} has been reached.")
                 should_continue = False
 
+        # Commit whatever records we were able to process
         self.commit_records()
 
+        # If errors were caught during processing, raise them now
+        if self.ingestion_errors:
+            # Log the affected query_params
+            bad_query_params = ", \n".join(
+                [f"{e.query_params}" for e in self.ingestion_errors]
+            )
+            logger.info(
+                "The following query_params resulted in errors: \n"
+                f"{bad_query_params}"
+            )
+            errors_str = "\n".join(
+                e.print_with_traceback() for e in self.ingestion_errors
+            )
+            raise AirflowException(
+                f"The following errors were encountered during ingestion:\n{errors_str}"
+            )
+
+    def get_query_params(
```

Comment:
I was attempting to avoid renaming the current `get_next_query_params` method, but I'm not happy with this naming. Any suggestions?

Reply:
Maybe `determine_query_params`?